internal/network/manifests_generator.go

```diff
@@ -238,6 +240,135 @@ func (m *ManifestsGenerator) AddSchedulableMastersManifest(ctx context.Context,
 	return nil
 }
 
+const diskEncryptionManifest = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: {{ .ROLE }}-{{ .MODE }}
+  labels:
+    machineconfiguration.openshift.io/role: {{ .ROLE }}
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      luks:
+        - name: root
+          device: /dev/disk/by-partlabel/root
+          clevis:
+		  {{- if eq .MODE "tpm" }}
+            tpm2: true
+		  {{- else if eq .MODE "tang" }}
+            tang:
+            {{- range .TANG_SERVERS }}
+              - url: {{ .Url }}
+                thumbprint: {{ .Thumbprint }}
+            {{- end }}
+		  {{- end }}
+          options: [--cipher, aes-cbc-essiv:sha256]
+          wipeVolume: true
+      filesystems:
+        - device: /dev/mapper/root
+          format: xfs
+          wipeFilesystem: true
+          label: root
+{{- if eq .MODE "tang" }}
+  kernelArguments:
+    - rd.neednet=1
+{{- end }}`
+
+func (m *ManifestsGenerator) createDiskEncryptionManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster,
+	manifestParams map[string]interface{}, filename string) error {
+
+	log.Infof("Creating manifest to encrypt installation disk on %s nodes using %s encryption", manifestParams["ROLE"], manifestParams["ROLE"])
+
+	content, err := fillTemplate(manifestParams, diskEncryptionManifest, log)
+	if err != nil {
+		log.WithError(err).Error("Failed to parse disk encryption manifest's template")
+		return err
+	}
+
+	if err := m.createManifests(ctx, c, filename, content); err != nil {
+		log.WithError(err).Errorf("Failed to create manifest to encrypt installation disk")
+		return err
+	}
+
+	return nil
+}
+
+func (m *ManifestsGenerator) AddDiskEncryptionManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster) error {
+
+	if reflect.DeepEqual(c.DiskEncryption, &models.DiskEncryption{}) {
+		return nil
+	}
+
+	manifestParams := map[string]interface{}{}
+
+	switch c.DiskEncryption.Mode {
+
+	case models.DiskEncryptionModeTpmv2:
+
+		manifestParams["MODE"] = "tpm"
+
+	case models.DiskEncryptionModeTang:
+
+		tangServers, err := common.UnmarshalTangServers(c.DiskEncryption.TangServers)
+		if err != nil {
+			log.WithError(err).Error("faild to unmarshal tang_server from cluster object")
+			return err
+		}
+
+		manifestParams["MODE"] = "tang"
+		manifestParams["TANG_SERVERS"] = tangServers
+
+	default:
+
+		return errors.New("None of the encryption modes were specified")
+	}
+
+	switch c.DiskEncryption.EnableOn {
+
+	case models.DiskEncryptionEnableOnAll:
+
+		manifestParams["ROLE"] = "master"
+		filename := fmt.Sprintf("99-openshift-%s-%s-encryption.yaml", manifestParams["ROLE"], manifestParams["MODE"])
+		if err := m.createDiskEncryptionManifest(ctx, log, c, manifestParams, filename); err != nil {
```

Comment:
Consider passing the role instead of the filename to `m.createDiskEncryptionManifest` and move these 2 lines to that function:
manifestParams["ROLE"] =
filename :=

e.g.
if err := m.createDiskEncryptionManifest(ctx, log, c, manifestParams, "master"); err != nil {

Reply:
Fixed.